CppSamples/EditData/EditGeodatabaseWithTransactions/EditGeodatabaseWithTransactions.h

```diff
@@ -0,0 +1,125 @@
+  Q_PROPERTY(QStringList currentFeatureTypes READ currentFeatureTypes NOTIFY currentFeatureTypesChanged)
+
+public:
+  explicit EditGeodatabaseWithTransactions(QObject* parent = nullptr);
+  ~EditGeodatabaseWithTransactions() override;
+
+  static void init();
+
+  // Property getters
+  bool startEditingEnabled() const{return m_startEditingEnabled;}
+  bool stopEditingEnabled() const{return m_stopEditingEnabled;}
+  bool requireTransaction() const{return m_requireTransaction;}
+  QString messageText() const{return m_messageText;}
+  bool loadingVisible() const{return m_loadingVisible;}
+  QStringList availableTableNames() const{return m_availableTableNames;}
+  QString selectedTableName() const{return m_selectedTableName;}
+  QStringList currentFeatureTypes() const{return m_currentFeatureTypes;}
+
+public slots:
+  void beginTransaction();
+  void stopTransaction();
+  void commitTransaction();
+  void rollbackTransaction();
+  void cancelTransaction();
+  void setRequireTransaction(bool require);
+  void setSelectedTableName(const QString& tableName);
+  void addFeatureAtLocation(const QString& tableName, const QString& featureTypeName, const QPoint& location);
+  void updateFeatureTypesForSelectedTable(const QString& tableName);
+
+signals:
+  void mapViewChanged();
+  void startEditingEnabledChanged();
+  void stopEditingEnabledChanged();
+  void requireTransactionChanged();
+  void messageTextChanged();
+  void loadingVisibleChanged();
+  void availableTableNamesChanged();
+  void selectedTableNameChanged();
+  void currentFeatureTypesChanged();
+  void featureTypeSelectionRequested(int x, int y);
+  void commitDialogRequested();
+
+private:
+  Esri::ArcGISRuntime::MapQuickView* mapView() const;
+  void setMapView(Esri::ArcGISRuntime::MapQuickView* mapView);
+  void onGeodatabaseDoneLoading_(const Esri::ArcGISRuntime::Error& error);
+  void connectSignals();
+
+  void setMessageText(const QString& message);
+  void setLoadingVisible(bool visible);
+  void updateFeatureTypesForTable(Esri::ArcGISRuntime::GeodatabaseFeatureTable* table);
+  void showExtent();
+
+  Esri::ArcGISRuntime::Map* m_map = nullptr;
+  Esri::ArcGISRuntime::MapQuickView* m_mapView = nullptr;
+
+  const Esri::ArcGISRuntime::Envelope *m_extent = nullptr;
+  Esri::ArcGISRuntime::Geodatabase* m_geodatabase = nullptr;
+
+  // Store references to all available feature tables
+  QList<Esri::ArcGISRuntime::GeodatabaseFeatureTable*> m_allFeatureTables;
+  QMap<QString, Esri::ArcGISRuntime::GeodatabaseFeatureTable*> m_tablesByName;
+
+  // UI state properties
+  bool m_startEditingEnabled = false;
+  bool m_stopEditingEnabled = false;
+  bool m_requireTransaction = true;
+  QString m_messageText;
+  bool m_loadingVisible = true;
+  QStringList m_availableTableNames;
+  QString m_selectedTableName;
+  QStringList m_currentFeatureTypes;
+
+private slots:
+  void gdbTransactionStatusChanged();
```

Comment:
You can use slots, but most of the time we omit the slots syntax. Qt can now directly invoke any C++ method from signals, so the `slots` feature is mostly redundant.

I do recommend postfixing all your private methods with a trailing underscore.